ssl/d1_lib.c

```diff
@@ -246,16 +253,34 @@ void dtls1_start_timer(SSL *s)
     }
 #endif
 
-    /* If timer is not set, initialize duration with 1 second */
+    /*
+     * If timer is not set, initialize duration with 1 second or
+     * a user-specified value if the timer callback is installed.
+     */
     if (s->d1->next_timeout.tv_sec == 0 && s->d1->next_timeout.tv_usec == 0) {
-        s->d1->timeout_duration = 1;
+
+        if (s->d1->timer_cb != NULL)
+            s->d1->timeout_duration_ms = s->d1->timer_cb(s, 1000);
```

Comment:
I am not sure if I agree. The value passed in/out is the timer _interval_.
Typically the application will implement this callback function:

```
static unsigned timer_cb(SSL *s, unsigned timer)
{
    return timer;
}
```
( or return 2*timer to double the interval).


If the interval is 0 then the re-transmit will happen immediately after the first
DTLS packet is sent out and will completely flood the network.

I have tried to keep the new code with sane values, in order not to break
the current implementation.